internal/datastore/crdb/crdb.go

```diff
@@ -371,36 +403,43 @@ func (cds *crdbDatastore) Features(ctx context.Context) (*datastore.Features, er
 	streamCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	time.AfterFunc(1*time.Second, cancel)
-	_, err = cds.writePool.Exec(streamCtx, fmt.Sprintf(cds.beginChangefeedQuery, tableTuple, head))
-	if err != nil && errors.Is(err, context.Canceled) {
-		features.Watch.Enabled = true
-		features.Watch.Reason = ""
-	} else if err != nil {
-		features.Watch.Enabled = false
-		features.Watch.Reason = fmt.Sprintf("Range feeds must be enabled in CockroachDB and the user must have permission to create them in order to enable the Watch API: %s", err.Error())
-	}
+
+	_ = cds.writePool.ExecFunc(streamCtx, func(ctx context.Context, tag pgconn.CommandTag, err error) error {
```

Comment:
There's a comment further up, but basically there's no way to know if stream has failed or just never responds. So instead we cancel after a second to get a value out.

This doesn't propagate errors up because the assumption is that if the stream fails / doesn't respond, watch is not enabled. So we successfully determined the state of the feature.

Reply:
oh wow, ok!